src/storage/heap_file.c

```diff
@@ -3906,7 +3903,7 @@ heap_stats_sync_bestspace (THREAD_ENTRY * thread_p, const HFID * hfid, HEAP_HDR_
 
 	  free_space = spage_max_space_for_new_record (thread_p, pg_watcher.pgptr);
 
-	  if (free_space >= min_freespace && free_space > HEAP_DROP_FREE_SPACE)
```

Comment:
This has a point in that it's open to the change of `heap_stats_get_min_freespace`. If it's taken into consideration, then I'll follow.

Reply:
I think it's just function which returns the minimum space. so, I don't have a good idea how to change `heap_stats_get_min_freespace`.
If you have any good ideas, could you share them?

Reply:
I didn't mean that we should make a change to the `heap_stats_get_min_freespace`. What I meant is the condition is not "useless" and has a point.

The `heap_stats_sync_bestspace` doesn't know how the minimum space for statistics is calculated by the `heap_stats_get_min_freespace` function. It just considers both the minimum space and `HEAP_DROP_FREE_SPACE`. With your change, it doesn't consider "the minimum space". 
Looking into the function, it can't be larger than `HEAP_DROP_FREE_SPACE` for now, but the upper function doesn't need to know and should not know this. In the existing code, if the way we calculate the minimum space is changed, the `heap_stats_sync_bestspace` takes this into consideration, but it will not with your change. That's what I meant by `it's open to the change of heap_stats_get_min_freespace`.

Reply:
To justify the condition as "useless", I think we should ensure that "the minimum space for statistics" doesn't need to be taken into consideration in the view of the `heap_stats_sync_bestspace`. It might be true, but I don't know.

Reply:
I've certainly overlooked that aspect! I'll comment as opinions are being organized, but it might take a little while to check.

Reply:
@wrlawodms 
I have been asked to quickly address these heap refactoring issues. I will revisit these issues following the analysis of the heap manager.